lib/iris/coords.py

```diff
@@ -1604,12 +1701,21 @@ def bounds(self, bounds):
 
             # Ensure the array is read-only.
             bounds.flags.writeable = False
-
-        self._bounds = bounds
+            self._bounds_dm.data = bounds
 
     def is_monotonic(self):
         return True
 
+    def replace_points(self, points, dtype=None, fill_value=None):
```

Comment:
> a common gateway for those checks

I've done something slightly different to this, which should be applicable in the same sort of manner. Let me know what you think about this approach!